pinns/src/sysctl.c

```diff
@@ -0,0 +1,105 @@
+#define _GNU_SOURCE
+
+#include <string.h>
+#include <fcntl.h>
+
+#include "sysctl.h"
+#include "utils.h"
+
+static int separate_sysctl_key_value (char* sysctl_key_value, char** sysctl_key, char** sysctl_value);
+static int write_sysctl_to_file (char * sysctl_key, char* sysctl_value);
+const char *sysctl_delim = "+";
+
+int configure_sysctls (char * const sysctls) {
+  char* sysctl = strtok(sysctls, sysctl_delim);
+  char* key = NULL;
+  char* value = NULL;
+  while (sysctl) {
+    if (separate_sysctl_key_value (sysctl, &key, &value) < 0)
+      return -1;
+
+    if (write_sysctl_to_file (key, value) < 0)
+      return -1;
+	
+	sysctl = strtok (NULL, sysctl_delim);
+  }
+
+  return 0;
+}
+
+// sysctl_key_value should be in the form `'key=value'`
+static int separate_sysctl_key_value (char* sysctl_key_value, char** sysctl_key, char** sysctl_value)
```

Comment:
I think if you drop `sysctl_` prefix from all the variables here (not the function name though), it will result in a better looking code.